### DISCUSSION TITLE
Sporadic results

### DISCUSSION THREAD

**Discussion by BNWEIN** [alexjustesen/speedtest-tracker#1782] [Q&A]
hey all

I am getting really sporadic results when using this and i am unsure why.  When i run a speed test directly in my Unifi Security Gateway, its always consistent, showing around 910mbps, however, i had speedtest tracker to run every hour and at first was getting good results, but they have since dropped off.

My Docker Machine is under very light load so its not a machine issue
The machine is cabled directly into the network so i don't believe its a localised issue
Manually running the tests outside of "on the hour", i am still getting sporadic results (However, after reading some threads here, i have since changed my cron job to run at 22 mins past every hour to avoid congestion)

Could i be choosing bad servers to be running my speed tests again? I am using the "SPEEDTEST_SERVERS" environmental value, and have the following servers selected: - SPEEDTEST_SERVERS=47230,14679,33216,16152,24281

![image](https://github.com/user-attachments/assets/64e32c6c-6ada-4f1a-85cd-d2216118afd6)

![image](https://github.com/user-attachments/assets/54be6f6b-64c9-4703-87dc-d842b889fe1e)

**Comment by alexjustesen** [comment]
Couple of things to keep in mind, Unifi speedtests use their own service so it's possible they're more reliable in your area. I know mine seem to have less fluctuation as well, I suspect because it's a less used service.

![image](https://github.com/user-attachments/assets/6b7c2fb3-6d45-473e-8499-d17e814c7ea0)

I'd try choosing some different servers, if that doesn't help you narrow down the issue and you're using "off" times it could be congestion on you ISP's network.

**Comment by BNWEIN** [comment]
Thanks Alex.

How many servers do you recommend using?  As you can see here: SPEEDTEST_SERVERS=47230,14679,33216,16152,24281 i am using a fair few already? Some of which at times show the expected results but at other times dont. Very strange. Is this just ookla being a bit crap?

**Reply by alexjustesen** [reply]
Since it chooses servers at random I have 3 in mine, now if I want to take out variables in the process I'll pick one server and see if the speeds normalize.

**Reply by BNWEIN** [reply]
I was worried doing one every hour to the same server they may start to throttle me or something?

**Reply by alexjustesen** [reply]
You should be ok hourly
